packages/vscode-extension/src/debug/runIconHandler.ts

```diff
@@ -22,12 +33,51 @@ export async function selectAndDebug(args?: any[]): Promise<Result<null, FxError
 }
 
 export function registerRunIcon(): void {
-  ext.context.subscriptions.push(vscode.workspace.onDidOpenTextDocument(enableRunIcon));
-  ext.context.subscriptions.push(vscode.workspace.onDidChangeWorkspaceFolders(enableRunIcon));
-  enableRunIcon();
+  ext.context.subscriptions.push(
```

Comment:
Is it better to just set the status of whether project is valid teamsfx project and don't care about whether active text editor changed. Do you use onDidChangeActiveTextEditor() here for situations when user checkout code or something else?

Reply:
Yes, the user may change the project, and the run icon status will be changed.
When the project is open, the project status will be force checked